app/src/main/java/com/concordium/wallet/ui/seed/recover/private_key/PrivateKeyRecoverViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.concordium.wallet.ui.seed.recover.private_key
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.concordium.wallet.App
+import com.concordium.wallet.BuildConfig
+import com.concordium.wallet.core.authentication.Session
+import com.concordium.wallet.data.preferences.AuthPreferences
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class PrivateKeyRecoverViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val session: Session = App.appCore.session
+
+    private val _saveKeySuccess = MutableStateFlow(false)
+    val saveKeySuccess = _saveKeySuccess.asStateFlow()
+
+    private val _validate = MutableStateFlow(false)
+    val validate = _validate.asStateFlow()
+
+    var privateKey = MutableStateFlow("")
+
+    fun setPredefinedKeyForTesting(password: String) = viewModelScope.launch {
+        if (BuildConfig.DEBUG) {
+            setPrivateKey(
+                "575f0f919c99ed4b7d858df2aea68112292da4eae98e2e69410cd5283f3c727b282caeba754f815dc876d8b84d3339c6f74c4127f238a391891dd23c74892943",
+                password
+            )
+        }
+    }
+
+    fun validatePrivateKey() {
```

Comment:
It's a bad validation, it allows any 128-char long text without spaces to be used as a seed, which results in non-functional wallet. Please, check if the input can be decoded as a hex string and the decoded data length in bytes is 64.